SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityLeaderCache.java

```diff
@@ -47,15 +48,43 @@
   private Map<String, RequestUtilization> requestUtilizations;
   private Map<String, SingularityAgentUsageWithId> agentUsages;
 
+  private final Object syncObject = new Object();
+
   private volatile boolean active;
+  private volatile boolean bootstrapping;
 
   @Inject
   public SingularityLeaderCache() {
     this.active = false;
+    this.bootstrapping = false;
+  }
+
+  public synchronized void startBootstrap() {
+    bootstrapping = true;
   }
 
-  public void activate() {
+  public synchronized void activate() {
     active = true;
+    bootstrapping = false;
+    synchronized (syncObject) {
+      syncObject.notify();
+    }
```

Comment:
what would this do?

Reply:
bit of background on wait + notify in https://www.baeldung.com/java-wait-notify. Though realizing I probably need notifyAll here actually. Leaving as a TODO for later